src/script/engine.zig

```diff
@@ -93,56 +82,56 @@ pub const Engine = struct {
         }
     }
 
-    /// Execute a single opcode
-    ///
-    /// # Arguments
-    /// - `opcode`: The opcode to execute
-    ///
-    /// # Returns
-    /// - `EngineError`: If an error occurs during execution
     fn executeOpcode(self: *Engine, opcode: u8) !void {
         self.log("Executing opcode: 0x{x:0>2}\n", .{opcode});
-        try switch (opcode) {
-            0x00...0x4b => try self.pushData(opcode),
-            0x4c => try self.opPushData1(),
-            0x4d => try self.opPushData2(),
-            0x4e => try self.opPushData4(),
-            0x4f => try self.op1Negate(),
-            0x51...0x60 => try self.opN(opcode),
-            0x61 => try self.opNop(),
-            0x69 => try self.opVerify(),
-            0x6a => try self.opReturn(),
-            0x6d => try self.op2Drop(),
-            0x6e => try self.op2Dup(),
-            0x6f => try self.op3Dup(),
-            0x73 => self.opIfDup(),
-            0x74 => self.opDepth(),
-            0x75 => try self.opDrop(),
-            0x76 => try self.opDup(),
-            0x87 => try self.opEqual(),
-            0x88 => try self.opEqualVerify(),
-            0x8b => try arithmetic.op1Add(self),
-            0x8c => try arithmetic.op1Sub(self),
-            0x8f => try arithmetic.opNegate(self),
-            0x90 => try arithmetic.opAbs(self),
-            0x91 => try arithmetic.opNot(self),
-            0x92 => try arithmetic.op0NotEqual(self),
-            0x93 => try arithmetic.opAdd(self),
-            0x94 => try arithmetic.opSub(self),
-            0x9a => try arithmetic.opBoolAnd(self),
-            0x9b => try arithmetic.opBoolOr(self),
-            0x9c => try arithmetic.opNumEqual(self),
-            0x9d => try arithmetic.opNumEqualVerify(self),
-            0x9e => try arithmetic.opNumNotEqual(self),
-            0x9f => try arithmetic.opLessThan(self),
-            0xa0 => try arithmetic.opGreaterThan(self),
-            0xa1 => try arithmetic.opLessThanOrEqual(self),
-            0xa2 => try arithmetic.opGreaterThanOrEqual(self),
-            0xa3 => try arithmetic.opMin(self),
-            0xa4 => try arithmetic.opMax(self),
-            0xa5 => try arithmetic.opWithin(self),
-            0xa9 => try self.opHash160(),
-            0xac => try self.opCheckSig(),
+
+        if (getPushDataLength(opcode)) |length| {
+            try self.pushData(length);
+            return;
+        }
```

Comment:
Are we sure we want to keep this? Can you explain what it does and why it should be there?

Reply:
yeah this is a check for the `Opcodes 1-75 (0x01 - 0x4B)`" https://wiki.bitcoinsv.io/index.php/Pushdata_Opcodes
 because the enum isn't rangeable, so I handle it separately

Reply:
Okay I get it.

Could you rename `getPushDataLength` to something more explicit, like `isUnnamedPushNDataOpcode`


And add a lil comment ontop of this line explaining this, with the link to the doc